api/src/org/apache/cloudstack/ha/HAConfigManager.java

```diff
@@ -0,0 +1,96 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.ha;
+
+import com.cloud.dc.DataCenter;
+import com.cloud.org.Cluster;
+
+import java.util.List;
+
+/**
+ * @since 4.11
+ */
+public interface HAConfigManager {
+    /**
+     * Configures HA for a resource by accepting the resource type and HA provider
+     * @param resourceId the ID of the resource
+     * @param resourceType the type of the resource
+     * @param haProvider the name of the HA provider
+     */
+    boolean configureHA(Long resourceId, HAResource.ResourceType resourceType, String haProvider);
+
+    /**
+     * Enables HA for resource Id of a specific resource type
+     * @param resourceId the ID of the resource
+     * @param resourceType the type of the resource
+     * @return returns true on successful enable
+     */
+    boolean enableHA(Long resourceId, HAResource.ResourceType resourceType);
+
+    /**
+     * Disables HA for resource Id of a specific resource type
+     * @param resourceId the ID of the resource
+     * @param resourceType the type of the resource
+     * @return returns true on successful disable
+     */
+    boolean disableHA(Long resourceId, HAResource.ResourceType resourceType);
+
+    /**
+     * Enables HA across a cluster
+     * @param cluster the cluster
+     * @return returns operation success
+     */
+    boolean enableHA(final Cluster cluster);
```

Comment:
Could these method names refer to resource they are enabling/disabling HA on? i.e. `enableClusterHA`, `enableZoneHA`

Reply:
they may, however it is still readable as the input params maybe of Cluster/Host/Zone type.